bundles/org.eclipse.osgi.service.http.servlet.contract/META-INF/MANIFEST.MF

```diff
@@ -0,0 +1,18 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: JavaServlet OSGi Contract Provider for Equinox
+Bundle-SymbolicName: org.eclipse.osgi.service.http.servlet.contract;singleton:=true
+Automatic-Module-Name: org.eclipse.osgi.service.http.servlet.contract
+Bundle-Vendor: Eclipse.org - Equinox
+Bundle-Version: 1.0.0.qualifier
+Fragment-Host: 
+ jakarta.servlet-api;bundle-version="[4.0.0,5.0.0)",
+ javax.servlet-api;bundle-version="[3.0.0,5.0.0)",
+ org.eclipse.jetty.servlet-api;bundle-version="[4.0.0,5.0.0)"
```

Comment:
A fragment is only allowed to specify a single host.

Reply:
Ok, somehow I assume this is possible. At least the runtime didn't complain, but looking in the spec it does not look like this is permitted.
But ok, then we would have to select one of these three, in case we continue with adding this fragment. In that case it is probably the most future-proof to choose `jakarta.servlet-api`.

Reply:
@tjwatson are you sure? I also though this but remember I have seen such "multi-host" already somewhere else (sadly I can't remember where/when) and was surprised there as well...

Reply:
> @tjwatson are you sure? I also though this but remember I have seen such "multi-host" already somewhere else (sadly I can't remember where/when) and was surprised there as well...

Yes, I'm sure the spec `Fragment-Host` syntax only allows for a single host symbolic name to be defined:

https://docs.osgi.org/specification/osgi.core/8.0.0/framework.module.html#framework.module.fragmenthost

```
Fragment-Host       ::= bundle-description
bundle-description  ::= symbolic-name 
                            ( ';' parameter )* // See [1.3.2](https://docs.osgi.org/specification/osgi.core/8.0.0/framework.introduction.html#framework.general.syntax)
```

It is true a fragment may be attached to multiple host bundles, but that is only when there are multiple versions of a bundle are installed (with the same symbolic name):

https://docs.osgi.org/specification/osgi.core/8.0.0/framework.module.html#framework.module.fragmentbundles

> Fragments can be attached to multiple hosts with the same symbolic name but different versions. If multiple fragments with the same symbolic name match the same host, then the Framework must only select one fragment, this must be the fragment with the highest version.


> Ok, somehow I assume this is possible. At least the runtime didn't complain, but looking in the spec it does not look like this is permitted.

This is because Equinox just uses the first one listed:

https://github.com/eclipse-equinox/equinox/blob/0582356d5ab0bc359b91dc242a92ecfc767463ed/bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/container/builders/OSGiManifestBuilderFactory.java#L501

Reply:
@tjwatson is `osgi.support.multipleHosts=true` gone already?
https://www.eclipse.org/forums/index.php?t=msg&th=276829&goto=1052923&#msg_1052923

Reply:
Regardless of my answer (which I don't recall off the top of my head), that was only about resolve against multiple host versions (with same BSN).

Reply:
I see.. then I maybe has wrong memories here...